### PR TITLE
fix: fix backpack page selector

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?
Fix backpack page selector

## How to test the changes?
1. Launch the explorer using `&DISABLE_backpack_editor_v1&ENABLE_backpack_editor_v2`.
2. Open the backpack.
3. Check the pagination works correctly.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at e8ddff2</samp>

Refactored the page selector component and its related prefabs, scripts, and interfaces to improve its flexibility, usability, and performance. Renamed the `GridPageSelector` and `GridPageButton` prefabs to `PageSelector` and `PageSelectorButton`, and changed their internal logic and appearance. Removed unused code and simplified the methods for setting and changing the wearable pages in the backpack editor HUD. Adjusted the `guid` fields of the meta files to reflect the renaming and refactoring of the prefabs and scripts.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/WearableGridController.cs [hunk 1]
```diff
@@ -185,7 +185,8 @@ private async UniTask<int> RequestWearablesAndShowThem(int page, CancellationTok
                 currentWearables = wearables.Select(ToWearableGridModel)
                                             .ToDictionary(item => item.WearableId, model => model);
 
-                view.SetWearablePages(page, (totalAmount / PAGE_SIZE) + 1);
+                view.SetWearablePages((totalAmount + PAGE_SIZE - 1) / PAGE_SIZE);
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/WearableGridController.cs:188]: what is the gain of separating the operation in two methods?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/WearableGridControllerShould.cs [hunk 2]
```diff
@@ -444,7 +444,7 @@ public void ChangePageWhenViewRequestsIt()
                                         Arg.Any<string>(),
                                         Arg.Any<(NftOrderByOperation type, bool directionAscendent)?>());
 
-            view.Received(1).SetWearablePages(3, 4);
+            view.Received(1).SetWearablePages(4);
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/WearableGridControllerShould.cs:447]: we are not validating in the tests that the `GoToPage` method is being called

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/WearableGridController.cs (hunk 1 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/WearableGridController.cs]: Sorry, you're right... it was not needed. I had some problems to reset the page selector to the first page each time we change the filters and I finally left the separated flow. I have unified them again. Done here: https://github.com/decentraland/unity-renderer/pull/5119/commits/defc148a43876106380609d549752438a85841f9

File: unity-renderer/Assets/UIComponents/Scripts/Components/PageSelector/PageSelectorButtonComponentView.cs [hunk 3]
```diff
@@ -3,47 +3,44 @@
 using UnityEngine;
 using UnityEngine.UI;
 
-namespace UIComponents.Scripts.Components
+public class PageSelectorButtonComponentView : BaseComponentView
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/UIComponents/Scripts/Components/PageSelector/PageSelectorButtonComponentView.cs]: Done!: https://github.com/decentraland/unity-renderer/pull/5119/commits/4e1f243581b45d3ad915c4f6dc283fbae10f8ad1

File: unity-renderer/Assets/UIComponents/Scripts/Components/PageSelector/PageSelectorComponentView.cs [hunk 4]
```diff
@@ -3,147 +3,148 @@
 using UnityEngine;
 using UnityEngine.UI;
 
-namespace UIComponents.Scripts.Components
+public class PageSelectorComponentView : BaseComponentView
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/UIComponents/Scripts/Components/PageSelector/PageSelectorComponentView.cs]: Done!: https://github.com/decentraland/unity-renderer/pull/5119/commits/4e1f243581b45d3ad915c4f6dc283fbae10f8ad1

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/WearableGridControllerShould.cs (hunk 2 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/WearableGridControllerShould.cs]: The `GoToPage` function was removed in the first feedback 👍

**Review by lorux0 — APPROVED** [review]

**Review by davidejensen — APPROVED** [review]
Awesome, approved!

**Pull request merged**